### PR TITLE
refactor: add `docker-compose.yml`, fix README, optimize Dockerfile

### PR DESCRIPTION
**Pull request by roschaefer**
## Pullrequest
On linux machines is nice to have  a `docker-compose.yml` as it doesn't consume extra resources. There were several broken links in the README which I fixed. I also advise to put all diagrams into the repository itself and don't link to other repositories.

On my machine the docker image gets down to 275MB. The tool [dive](https://github.com/wagoodman/dive) tells me that were adding 158MB to the original alpine-node image (mostly `node_modules`).

I also removed `Dockerfile.dev` as it's better to specify build targets in `docker-compose.yml`.

### Issues

- [X] None

### Checklist

- [X] None

### How2Test

- [X] None

### Todo

- [X] None

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,57 +2,60 @@
...
-```
+### Configuration
 
-### Test Project
+### With docker
+
+Run:
 ```
-yarn lint
+docker-compose up
```
Comment (ManAnRuck) [inline README.md:42]: It looks like docker changed the command to use `docker compose`
```
Docker Compose is now in the Docker CLI, try `docker compose up`
```

**Review by ManAnRuck — CHANGES_REQUESTED** [review]
I get the following error on run
```
cd bundestag.io-api
docker compose up
```
Bildschirmfoto 2021-05-22 um 12 39 58

Docker version 20.10.6, build 370c289

**Review by ManAnRuck — COMMENTED** [review]
Do we want to have a docker-compose file for **every service**, **for project** or **both**?
Especially when we thin about to convert the git-submodule structure to a monorepo.
Any thoughts about it?

**Comment by roschaefer** [comment]
> I get the following error on run
> 
> ```
> cd bundestag.io-api
> docker compose up
> ```
> 
> Bildschirmfoto 2021-05-22 um 12 39 58
> 
> Docker version 20.10.6, build 370c289

This is super weird. Maybe your docker version complaints about UPCASE written stage names? I wrote it like that because it was upcase before. For me it's no problemo to write the stage names in all-lowercase letters.

File: README.md (hunk 1 above)
Comment (roschaefer) [reply README.md:42]: I'm on Archlinux and `docker --version` `20.10.6`. When I `docker compose up` without the dash I get:
```
docker: 'compose' is not a docker command.
See 'docker --help'
```

So maybe this is a Mac OS specific subcommand?

**Comment by roschaefer** [comment]
> Do we want to have a docker-compose file for **every service**, **for project** or **both**?
> Especially when we thin about to convert the git-submodule structure to a monorepo.
> Any thoughts about it?

From my experience it's much easier to have all `docker-compose.yml` in *one folder*. However, I suggest we migrate step-by-step. Right now, there is only a `docker-compose.yml` for this particular service, so that's why it's located in this repository. Once we remove git submodules, we can move the `docker-compose.yml` to the parent repository.

File: README.md (hunk 1 above)
Comment (ManAnRuck) [reply README.md:42]: yes maybe. both works for me, so lets let it be like it is .)

**Comment by ManAnRuck** [comment]
> > I get the following error on run
> > ```
> > cd bundestag.io-api
> > docker compose up
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Bildschirmfoto 2021-05-22 um 12 39 58
> > Docker version 20.10.6, build 370c289
> 
> This is super weird. Maybe your docker version complaints about UPCASE written stage names? I wrote it like that because it was upcase before. For me it's no problemo to write the stage names in all-lowercase letters.

yes that's weird! now it work's with lowercase :)

**Comment by roschaefer** [comment]
This is what I want to post to this PR on Friday, but it seems not to have been uploaded successfully:
**Before**
swappy-20210520_215022
**After**
swappy-20210520_214829

**Review by ManAnRuck — APPROVED** [review]

File: docker-compose.override.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+version: "3.9"
```
Comment (ManAnRuck) [inline docker-compose.override.yml:1]: What does this file do? How to execute it?

Comment (roschaefer) [reply docker-compose.override.yml:1]: > By default, Compose reads two files, a docker-compose.yml and an optional docker-compose.override.yml file. By convention, the docker-compose.yml contains your base configuration. 

See: https://docs.docker.com/compose/extends/

So you could e.g. run `docker-compose up` for development and `docker-compose -f docker-compose.yml up` to simulate production.

Comment (roschaefer) [reply docker-compose.override.yml:1]: By the way: Who is going to merge? The reviewer or the author?

**Pull request merged**